Assignment4/carlosferrero-upm+180066/task07.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1OdhDD8AEKIOdW7LgURWgWhd3__-JQAqK
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+for s, p, o in g:
+  print(s,p,o)
+
+from rdflib.plugins.sparql import prepareQuery
+NS = Namespace("http://somewhere#")
+ORG = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+q1 = prepareQuery('''
+  SELECT ?Subject  WHERE {
+    ?Subject org:subClassOf ns:Person.
+  }  
+  '''
+    ,
+  initNs = { "ns": NS, "org" : ORG}
+)
+
+# Visualize the results
+for r in g.query(q1):
+  print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+NS = Namespace("http://somewhere#")
+ORG = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+ORG2 = Namespace("http://www.w3.org/1999/02/22-rdf-syntax-ns#")
+q2 = prepareQuery('''
+  SELECT DISTINCT ?Instance  WHERE {
+     {?Subject org:subClassOf ns:Person.
+     ?Instance ?X ?Subject.}
+      UNION
+     {?Instance ?X ns:Person.} 
```

Comment:
?x should be rdf:type .otherwise you are retrieving all the things that have any relationship with person